onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc

```diff
@@ -47,6 +47,14 @@ namespace onnxruntime {
 namespace contrib {
 namespace GenerationCudaDeviceHelper {
 
+// This function assumes the Attention type is same as graph's past state type.
+// e.g In the case of past(fp32) -> cast to fp16 -> Attention(fp16), the reorder
+// function will use the fp32 chunk size and cause the model silently generates
+// the incorrect results.
+// TODO: Fix this issue. Either retrive the Attention op type from the graph or
```

Comment:
>TODO [](http://example.com/codeflow?start=3&length=4)

We need to experiment how the decoder_mask_attention kernel perform if we don't introduce this reorder.